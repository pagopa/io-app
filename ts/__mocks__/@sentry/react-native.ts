export const captureException = jest.fn();
export const captureMessage = jest.fn();
export const captureEvent = jest.fn();
export const setContext = jest.fn();
export const setExtras = jest.fn();
export const setExtra = jest.fn();
export const setTags = jest.fn();
export const setTag = jest.fn();
export const setUser = jest.fn();
export const lastEventId = jest.fn();
export const captureCheckIn = jest.fn();
export const withMonitor = jest.fn();
export const flush = jest.fn().mockResolvedValue(false);
export const close = jest.fn().mockResolvedValue(false);
export const isInitialized = jest.fn().mockReturnValue(false);
export const isEnabled = jest.fn().mockReturnValue(false);
export const addEventProcessor = jest.fn();
export const startSession = jest.fn();
export const endSession = jest.fn();
export const captureSession = jest.fn();
export const ErrorBoundary = jest.fn(({ children }) => children);
