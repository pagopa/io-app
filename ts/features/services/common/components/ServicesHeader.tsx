import {
  Avatar,
  H3,
  IOColors,
  IOSpacingScale,
  IOStyles,
  IOVisualCostants,
  LabelSmall,
  useIOTheme,
  VSpacer
} from "@pagopa/io-app-design-system";
import React from "react";
import { ImageURISource, StyleSheet, View } from "react-native";
import Placeholder from "rn-placeholder";

const ITEM_PADDING_VERTICAL: IOSpacingScale = 6;
const AVATAR_MARGIN_RIGHT: IOSpacingScale = 16;

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    paddingVertical: ITEM_PADDING_VERTICAL
  },
  itemAvatar: {
    marginRight: AVATAR_MARGIN_RIGHT
  }
});

export type ServicesHeaderProps = {
  logoUri: ReadonlyArray<ImageURISource>;
  title: string;
  subTitle: string;
};

export const ServicesHeader = ({
  logoUri,
  title,
  subTitle
}: ServicesHeaderProps) => {
  const theme = useIOTheme();

  return (
    <View style={styles.container} testID="services-header">
      <View style={styles.itemAvatar}>
        <Avatar logoUri={logoUri} size="medium" />
      </View>
      <View style={IOStyles.flex}>
        <H3 color={theme["textHeading-secondary"]}>{title}</H3>
        <LabelSmall color="grey-700" weight="Bold">
          {subTitle}
        </LabelSmall>
      </View>
    </View>
  );
};

export const ServicesHeaderSkeleton = () => (
  <View
    style={styles.container}
    accessible={true}
    testID="services-header-skeleton"
  >
    <View style={styles.itemAvatar}>
      <Placeholder.Box
        width={IOVisualCostants.avatarSizeMedium}
        height={IOVisualCostants.avatarSizeMedium}
        animate="fade"
        radius={IOVisualCostants.avatarRadiusSizeMedium}
        color={IOColors["grey-200"]}
      />
    </View>
    <View style={IOStyles.flex}>
      <Placeholder.Box
        animate="fade"
        color={IOColors["grey-200"]}
        height={16}
        width={"100%"}
        radius={8}
      />
      <VSpacer size={8} />
      <Placeholder.Box
        animate="fade"
        color={IOColors["grey-200"]}
        height={16}
        width={"80%"}
        radius={8}
      />
      <VSpacer size={8} />
      <Placeholder.Box
        animate="fade"
        color={IOColors["grey-200"]}
        height={8}
        width={"60%"}
        radius={8}
      />
    </View>
  </View>
);
