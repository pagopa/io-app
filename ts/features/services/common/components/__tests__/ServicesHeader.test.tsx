import React from "react";
import { render } from "@testing-library/react-native";
import { ServicesHeader } from "../ServicesHeader";

describe("ServicesHeader component", () => {
  it("should match the snapshot", () => {
    const component = render(
      <ServicesHeader
        logoUri={require("../../../../../../img/test/logo.png")}
        title={"#### title ####"}
        subTitle={"#### subTitle ####"}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
