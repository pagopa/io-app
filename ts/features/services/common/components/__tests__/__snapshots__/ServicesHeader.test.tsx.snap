// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesHeader component should match the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingVertical": 6,
    }
  }
  testID="services-header"
>
  <View
    style={
      {
        "marginRight": 16,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        [
          {
            "borderColor": "rgba(14,15,19,0.1)",
            "borderCurve": "continuous",
            "borderWidth": 1,
            "overflow": "hidden",
          },
          {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 12,
            "height": 66,
            "padding": 9,
            "width": 66,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#FFFFFF",
              "borderCurve": "continuous",
              "overflow": "hidden",
            },
            {
              "borderRadius": 3,
            },
          ]
        }
      >
        <Image
          accessibilityIgnoresInvertColors={true}
          onError={[Function]}
          source={
            {
              "testUri": "../../../img/test/logo.png",
            }
          }
          style={
            {
              "height": "100%",
              "resizeMode": "contain",
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      color="bluegreyDark"
      defaultColor="bluegreyDark"
      defaultWeight="Semibold"
      dynamicTypeRamp="title2"
      font="TitilliumSansPro"
      fontStyle={
        {
          "fontSize": 22,
          "lineHeight": 33,
        }
      }
      maxFontSizeMultiplier={1.25}
      style={
        [
          {
            "fontSize": 22,
            "lineHeight": 33,
          },
          {
            "color": "#17324D",
            "fontFamily": "Titillium Sans Pro",
            "fontStyle": "normal",
            "fontWeight": "600",
          },
        ]
      }
      weight="Semibold"
    >
      #### title ####
    </Text>
    <Text
      allowFontScaling={false}
      color="grey-700"
      defaultColor="blue"
      defaultWeight="Bold"
      dynamicTypeRamp="footnote"
      font="TitilliumSansPro"
      fontSize="regular"
      fontStyle={
        {
          "fontSize": 14,
          "lineHeight": 21,
        }
      }
      maxFontSizeMultiplier={1.25}
      style={
        [
          {
            "fontSize": 14,
            "lineHeight": 21,
          },
          {
            "color": "#555C70",
            "fontFamily": "Titillium Sans Pro",
            "fontStyle": "normal",
            "fontWeight": "700",
          },
        ]
      }
      weight="Bold"
    >
      #### subTitle ####
    </Text>
  </View>
</View>
`;
